platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationAdminMigratable.java

```diff
@@ -169,9 +199,99 @@ private ImportMigrationConfigurationAdminEntry updateNecessaryConfigurationAdmin
     if (entry.getPid().equals("org.codice.ddf.security.filter.login.Session")) {
       entry.setPid("ddf.security.http.impl.HttpSessionFactory");
     }
+    // this pid was updated in version 2.17.0
+    // see https://github.com/codice/ddf/pull/5105
+    if (entry.getPid().equals("ddf.security.sts.guestclaims")) {
+      entry.setPid("ddf.security.guest.realm");
+    }
+    // this pid was updated in version 2.17.0
+    // see https://github.com/codice/ddf/pull/5105
+    if (entry.getPid().equals(WEB_CONTEXT_POLICY_MANAGER_PID)) {
+      entry = updateWebContextPolicyManagerConfiguration(entry);
+    }
+    return entry;
+  }
+
+  private ImportMigrationConfigurationAdminEntry updateWebContextPolicyManagerConfiguration(
+      ImportMigrationConfigurationAdminEntry entry) {
+    Map<String, Object> defaults = getDefaultProperties(WEB_CONTEXT_POLICY_MANAGER_PID);
+    Dictionary<String, Object> props = entry.getProperties();
+    props.remove("realms");
+
+    String[] whiteListContexts = (String[]) props.get(WHITE_LISTED_CONTEXTS);
+    String[] whiteListContextDefaults = (String[]) defaults.get(WHITE_LISTED_CONTEXTS);
+
+    Set<String> whiteListContextsSet =
+        Stream.of(whiteListContexts, whiteListContextDefaults)
+            .flatMap(Stream::of)
+            .collect(Collectors.toSet());
+
+    props.put(WHITE_LISTED_CONTEXTS, String.join(",", whiteListContextsSet));
+
+    String[] authenticationTypes = (String[]) props.get(AUTHENTICATION_TYPES);
+    List<String> authenticationTypesList = new ArrayList<>(Arrays.asList(authenticationTypes));
+
+    for (int i = 0; i < authenticationTypesList.size(); i++) {
+      String val = authenticationTypesList.get(i);
+      String[] parts = val.split("=");
+
+      if (parts.length == 2) {
+        String[] auths = parts[1].split("\\|");
+        List<String> list =
+            Arrays.stream(auths)
+                .filter(a -> !StringUtils.containsIgnoreCase(a, "guest"))
+                .map(a -> a.replaceAll("(?i)idp", "SAML"))
+                .collect(Collectors.toList());
+        authenticationTypesList.set(i, String.format("%s=%s", parts[0], String.join("|", list)));
+      }
+    }
+
+    props.put(AUTHENTICATION_TYPES, String.join(",", authenticationTypesList));
+
+    if (props.get(GUEST_ACCESS) == null) {
+      props.put(GUEST_ACCESS, "true");
```

Comment:
✏️ I just realized that this is technically a hard-coded value. If they change the default value, this will stay the same. Instead we should get the default value for this property, which will be fairly easy. I think we should do the same for the `SESSION_ACCESS` one.